pylxd/client.py

```diff
@@ -349,6 +350,34 @@ def received_message(self, message):
         self.messages.append(json_message)
 
 
+# Helper function used by Client.authenticate()
+def _is_a_token(secret):
+    """Inspect the provided secret to determine if it is a trust token.
+
+    Try to base64 decode and parse the JSON to see if it contains a "secret" key.
+
+    :param secret: The secret to inspect
+    :type secret: str
+    :returns: True if the secret is a trust token
+
+    >>> _is_a_token("password")
+    False
+    >>> _is_a_token(base64.b64encode("password".encode("utf-8")))
+    False
+    >>> token = '{"client_name":"foo","fingerprint":"abcd","addresses":["192.0.2.1:8443"],"secret":"I-am-a-secret","expires_at":"0001-01-01T00:00:00Z","type":""}'
+    >>> _is_a_token(base64.b64encode(json.dumps(token).encode("utf-8")))
+    True
+    """
+    try:
+        b64 = base64.b64decode(secret)
+        token = json.loads(b64.decode("utf-8"))
+        return "secret" in token
+    except (TypeError, ValueError, json.JSONDecodeError, base64.binascii.Error):
+        pass
+
+    return False
```

Comment:
I'm hesitant here with the style, @hamistao opinions?:
```suggestion
    except (TypeError, ValueError, json.JSONDecodeError, base64.binascii.Error):
        return False
```

Reply:
I like your suggestion better, but both are valid in my view